### PR TITLE
docs: Add section from empty workspaces migration

### PR DESCRIPTION
**Pull request by frascuchon**
This is a missing section for migrating datasets with empty workspaces. It will be mandatory for the next release.

This section will be referenced in the next release, once this PR #2373  is included.

### CODE REVIEW

File: docs/_source/guides/log_load_and_prepare_data.ipynb [hunk 1]
```diff
@@ -411,15 +411,81 @@
...
   },
   {
-   "attachments": {},
    "cell_type": "markdown",
-   "id": "289724c8",
-   "metadata": {},
+   "source": [
+    "## Migrating dataset from \"emtpy\" workspaces\n",
+    "\n",
+    "For old argilla versions, datasets could be created with no workspace inf (`-` workspace in UI).\n",
```
Comment (davidberenstein1957) [inline docs/_source/guides/log_load_and_prepare_data.ipynb]:
```suggestion
    "For argilla versions `<1.4.0`, datasets could be created without a workspace.\n",
```

File: docs/_source/guides/log_load_and_prepare_data.ipynb [hunk 2]
```diff
@@ -411,15 +411,81 @@
...
+    "\n",
+    "New versions expect to work with datasets in a given workspace.\n",
```
Comment (davidberenstein1957) [inline docs/_source/guides/log_load_and_prepare_data.ipynb]:
```suggestion
    "Newer versions (>=1.4.0), expect to work with datasets assigned to a specific workspace.\n",
```

File: docs/_source/guides/log_load_and_prepare_data.ipynb [hunk 3]
```diff
@@ -411,15 +411,81 @@
...
+    "\n",
+    "If you have datasets with empty workspaces, you should move them to a specific workspace to keep your workflow compatible with the latest versions\n",
```
Comment (davidberenstein1957) [inline docs/_source/guides/log_load_and_prepare_data.ipynb]:
```suggestion
    "If you have datasets within an empty workspaces, you should move them to a specific workspace to keep your workflow compatible with the latest versions\n",
```

File: docs/_source/guides/log_load_and_prepare_data.ipynb [hunk 4]
```diff
@@ -411,15 +411,81 @@
...
+    "\n",
+    "Here is a code snippet showing how to do it\n",
+    "\n",
+    "<div class=\"alert alert-warning\">\n",
+    "\n",
+    "Warning\n",
+    "\n",
+    "From argilla version v1.4.0, datasets with empty workspace won't be supported, and you should migrate them before upgrading argilla.\n",
```
Comment (davidberenstein1957) [inline docs/_source/guides/log_load_and_prepare_data.ipynb]:
```suggestion
    "From argilla version v1.4.0, datasets with empty workspace won't be supported anymore, and you should migrate them before upgrading argilla.\n",
```

**Comment by davidberenstein1957** [comment]
@frascuchon do we want to start using `rg = Argilla()`? Otherwise `import argilla as rg` or `rg.init` are more well-known.

**Comment by frascuchon** [comment]
Okay, I will change the code snippet using more popular functions

**Pull request merged**